apps/system/js/usb_storage.js

```diff
@@ -200,35 +298,54 @@
      * @memberof UsbStorage.prototype
      * @param {Number} val The value we are setting automount to.
      */
-    _setMode: function(val) {
+    _setMode: function(mode) {
+      DEBUG && debug("_setMode: setting mode to " + this._modeStr(mode));
       var param = {};
-      param[this.umsMode] = val;
+      param[this.umsMode] = mode;
+      this._mode = mode;
       SettingsListener.getSettingsLock().set(param);
     },
 
     /**
-     * General event handler interface.
-     * Updates the overlay with as we receive load events.
+     * Sets the mode to agree with the current state of things.
      * @memberof UsbStorage.prototype
-     * @param  {DOMEvent} evt The event.
      */
-    handleEvent: function(e) {
-      switch (e.type) {
-        case 'lockscreen-appopened':
-          // Setting mode due to screen locked
-          this._setMode(this.automounterDisableWhenUnplugged);
-          break;
-        case 'lockscreen-appclosed':
-          if (typeof(this._mode) == 'undefined') {
-            return;
-          }
+    _updateMode: function() {
+      if (typeof(this._enabled) == 'undefined') {
+        DEBUG && debug("_updateMode: _enabled not yet set - ignoring");
+        return;
+      }
+      if (typeof(this._locked) == 'undefined') {
+        DEBUG && debug("_updateMode: _locked not yet set - ignoring");
+        return;
+      }
+      if (typeof(this._lockEnabled) == 'undefined') {
+        DEBUG && debug("_updateMode: _lockEnabled not yet set - ignoring");
+        return;
+      }
 
-          // Setting mode due to screen unlocked
-          this._setMode(this._mode);
-          break;
-        default:
-          return;
+      var mode = this.automounterDisable;
+
+      if (this._enabled && (!this._lockEnabled || !this._locked)) {
```

Comment:
ok - So then the lockappopened/closed thing can be used as a trigger point to check for changes in Service.locked and I can just poll Service.locked at start time to cover off startup conditions.

I'll put back some of the code I deleted to make it work that way. I need to get similar information for controlling adb, so in a separate bug, I'll also rework b2g/chrome/content/devtools/adb.js

Right now, I've been burned several times by the lock API changing and not knowing about it, so the USB and adb stuff keeps getting broken. Is there a way we can try to prevent ths in the future?